app/src/main/java/com/team/uptech/pomodoro/presentation/presenter/impl/MainPresenterImpl.kt

```diff
@@ -0,0 +1,117 @@
+package com.team.uptech.pomodoro.presentation.presenter.impl
+
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.team.uptech.pomodoro.dagger.scope.PerActivity
+import com.team.uptech.pomodoro.data.model.Pomodoro
+import com.team.uptech.pomodoro.domain.interactor.StartTimerUseCase
+import com.team.uptech.pomodoro.domain.interactor.TimerUseCase
+import com.team.uptech.pomodoro.presentation.presenter.MainPresenter
+import com.team.uptech.pomodoro.presentation.service.TimerService
+import com.team.uptech.pomodoro.presentation.ui.view.MainView
+import io.reactivex.android.schedulers.AndroidSchedulers
+import javax.inject.Inject
+
+/**
+ * Created on 27.04.17.
+ */
+@PerActivity
+class MainPresenterImpl @Inject constructor(val context: Context,
+                                            val startTimerUseCase: StartTimerUseCase,
+                                            val timerUseCase: TimerUseCase) : MainPresenter {
+
+    private var mainView: MainView? = null
+
+    override fun bind(view: MainView) {
+        this.mainView = view
+    }
+
+    override fun unbind() {
+        this.mainView = null
+    }
+
+    override fun onStartClicked() {
+        startTimerUseCase.changePomodoroType()
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnSubscribe { mainView?.showProgress() }
+                .doAfterTerminate { mainView?.hideProgress() }
+                .doAfterSuccess {
```

Comment:
why do you do stuff in the doAfterSuccess() and not in the subscribe()'s success block? this seems really weird to me